modules/system-info/reporters/server.php

```diff
@@ -410,4 +403,63 @@ public function get_elementor_library() {
 			'value' => 'Connected',
 		];
 	}
+
+	/**
+	 * Get permissions by keys. If empty keys, get all permissions.
+	 * Be aware that you can only use key paths that are defined in the get_paths() method.
+	 *
+	 * @param array $key_paths
+	 * @return array {read: bool, write: bool, execute: bool}
+	 */
+	public function get_permissions( array $key_paths = [] ):array {
+		$paths = $this->get_paths();
+		$all_key_paths = array_keys( $paths );
+
+		$key_paths = empty( $key_paths ) ? $all_key_paths : array_intersect( $key_paths, $all_key_paths );
+
+		$permissions = [];
+
+		foreach ( $key_paths as $key_path ) {
+			$path = $paths[ $key_path ];
+
+			$permissions[ $key_path ] = $this->get_path_permissions( $path );
+		}
+
+		return $permissions;
+	}
+
+	/**
+	 * Get all the paths that we might need to check permissions.
+	 *
+	 * @return array
+	 */
+	private function get_paths(): array {
```

Comment:
If we're gonna use the method I said above, maybe change it to:
```suggestion
	private function get_system_paths(): array {
```

and it can also be public, might be useful

Reply:
Done